infrastructure/index.ts

```diff
@@ -257,47 +242,121 @@ function getMimeType(filePath: string): string | undefined {
-                    source: new pulumi.asset.FileAsset(filePath),
-                    contentType: getMimeType(filePath) || undefined,
-                },
-                {
-                    parent: contentBucket,
-                });
-        } else {
-            const s3Redirect = new aws.s3.BucketObject(
-                relativeFilePath,
-                {
-                    ...baseArgs,
-                    source: new pulumi.asset.FileAsset("/dev/null"), // Empty file.
-                    websiteRedirect: translateRedirect(filePath, redirect),
-                },
-                {
-                    parent: contentBucket,
-                });
-        }
+// Certain files, like JavaScript and CSS files produced during the build, must exist
+// before the website's HTML files can use them (otherwise, pages won't render properly,
+// or may be unusable), and others, like fonts and images, would be nice to have there in
+// advance, too. Extensions listed here are used for identifying files that should exist
+// on S3 before any HTML pages are updated to reference them.
+const assetTypes = [
+    "js",
+    "css",
+    "png",
+    "jpg",
+    "gif",
+    "svg",
+    "pdf",
+    "eot",
+    "ttf",
+    "woff",
+    "woff2",
+];
+
+// Keep track of the asset paths we encounter and the resources we create from them, so we
+// can declare the necessary dependency relationships.
+const assetPaths: string[] = [];
+const assetResources: aws.s3.BucketObject[] = [];
+
+// Find all files matching the patterns above, creating a new BucketObject for each one.
+// These objects are declared first in order to pass them conditionally as dependencies
+// below.
+assetTypes.forEach(pattern => {
+    glob.sync(`${webContentsRootPath}/**/*.${pattern}`).forEach(assetPath => {
+
+        // Remember the asset path, so we can ignore it later.
+        assetPaths.push(assetPath);
+
+        // Turn the full (local) path into a relative path, so we can use it for an S3 object key.
+        const relativeAssetPath = assetPath.replace(webContentsRootPath + "/", "");
+
+        // Create a resource object with the asset.
+        const assetResource = new aws.s3.BucketObject(
+            relativeAssetPath,
+            {
+                acl: "public-read",
+                key: relativeAssetPath,
+                source: new pulumi.asset.FileAsset(assetPath),
+                contentType: getMimeType(assetPath) || undefined,
+                bucket: contentBucket,
+            },
+            {
+                parent: contentBucket,
+            },
+        );
+
+        // Add the bucket resource to the list of asset dependencies.
+        assetResources.push(assetResource);
     });
+});
+
+// Now create bucket objects for everything else, bypassing anything created already.
+glob.sync(`${webContentsRootPath}/**/*`).forEach(filePath => {
+
+    // Ignore directories.
+    if (fs.lstatSync(filePath).isDirectory()) {
+        return;
+    }
```

Comment:
Instead of this check, wondering if we should let `glob` handle it via the `nodir` option, e.g.:

```typescript
glob.sync(`${webContentsRootPath}/**/*`, { nodir: true }).forEach(filePath => {
```

https://github.com/isaacs/node-glob#options

> * `nodir` Do not match directories, only files. (Note: to match only directories, simply put a `/` at the end of the pattern.)

Reply:
That works well -- and there's a `dot` boolean that handles the `.net` matching issue as well. 🎉